src/loader/bgw_launcher.c

```diff
@@ -588,25 +588,55 @@ launcher_handle_message(HTAB *db_htab)
 	return true;
 }
 
+/* ordered by priority, SIGTERM should take precedence */
+typedef enum ShutdownSignal
+{
+	NO_SIGNAL = 0,
+	GOT_SIGINT,
+	GOT_SIGTERM,
+}			ShutdownSignal;
+
+static volatile sig_atomic_t shutdown_signal = NO_SIGNAL;
```

Comment:
For when coverity complains about these line: yes, this is probably the safest way to set a signal-variable; if we really don't want to use an `enum` we can just `#define` the constants, for the numbers we're using it should amount to the same thing